source/Engine.cpp

```diff
@@ -1261,20 +1261,27 @@ void Engine::EnterSystem()
 				CreateWeather(hazard, stellar.Position());
 	}
 
-	const Fleet *raidFleet = system->GetGovernment()->RaidFleet();
-	const Government *raidGovernment = raidFleet ? raidFleet->GetGovernment() : nullptr;
-	if(raidGovernment && raidGovernment->IsEnemy())
-	{
-		pair<double, double> factors = player.RaidFleetFactors();
-		double attraction = .005 * (factors.first - factors.second - 2.);
-		if(attraction > 0.)
-			for(int i = 0; i < 10; ++i)
-				if(Random::Real() < attraction)
-				{
-					raidFleet->Place(*system, newShips);
-					Messages::Add("Your fleet has attracted the interest of a "
-							+ raidGovernment->GetName() + " raiding party.", Messages::Importance::Highest);
-				}
+	for(const auto &raidFleet : system->GetGovernment()->RaidFleets())
+	{
+		const Government *raidGovernment = raidFleet.first->GetGovernment();
+		if(raidGovernment && raidGovernment->IsEnemy())
+		{
+			pair<double, double> factors = player.RaidFleetFactors();
+			double attraction = .005 * (factors.first - factors.second - raidFleet.second);
+			// Consider the other fleets in the local system.
+			// Do not take into account the raid fleet, that would just mess with the minimum attraction of it.
+			for(const auto &fleet : system->Fleets())
+				attraction -= sqrt(fleet.Get()->Strength() / 10000.) / fleet.Period()
+					* (fleet.Get()->GetGovernment()->IsEnemy(raidGovernment) ? 1. : -1.);
```

Comment:
Do you have any numbers on why this is the way you've done it? How much less likely is it to be attacked in Sol? How much more likely is it to be attacked in Shaula?

Why a multiplier of -1 for fleets that aren't enemies instead of 0? By using -1, fleets that are neutral to the raid fleet but might also be neutral to the player increase the change of getting jumped.

Reply:
Since this is based on the spawnrate and not the actual fleets currently there so this can easily be calculated, I didnt do it though
About the 1 0 -1 I'll make it elegant

Reply:
Only question that remains is how much it should matter, rn I made it `(raidStrength / (fleet.Get()->Strength() / fleet.Period()))` so it wont matter too much but mb we should multiply the other fleet by smth
could always be tuned in a later PR, and eventually be made part of the gamerules

Reply:
Flip it to `fleet strength / raid strength`. A stronger system fleet should *increase* the multiplier on how hard attraction is hit, not decrease it.